src/Http/Http.Extensions/src/HttpResponseJsonExtensions.cs

```diff
@@ -89,13 +90,23 @@ public static partial class HttpResponseJsonExtensions
 
         response.ContentType = contentType ?? JsonConstants.JsonContentTypeWithCharset;
 
+        var startTask = Task.CompletedTask;
+        if (!response.HasStarted)
+        {
+            // Flush headers before starting Json serialization. This avoids an extra layer of buffering before the first flush.
+            startTask = response.StartAsync(cancellationToken);
+        }
+
         // if no user provided token, pass the RequestAborted token and ignore OperationCanceledException
-        if (!cancellationToken.CanBeCanceled)
+        if (!startTask.IsCompleted || !cancellationToken.CanBeCanceled)
         {
-            return WriteAsJsonAsyncSlow(response.Body, value, options, response.HttpContext.RequestAborted);
+            return WriteAsJsonAsyncSlow(startTask, response.BodyWriter, value, options,
+                ignoreOCE: !cancellationToken.CanBeCanceled,
+                cancellationToken.CanBeCanceled ? cancellationToken : response.HttpContext.RequestAborted);
```

Comment:
The "Slow" path is normal one right, since [the middleware](https://github.com/aspnet/Benchmarks/blob/40b0d04a53323fce8fcbfcbba297ebba16eee1f5/src/Benchmarks/Middleware/JsonMiddleware.cs#L37) doesn't pass a cancellation token?

I think it would be better to change `WriteAsJsonAsync` not pass in `RequestAborted` by default similar to `WriteAsync`. Essentially serializing JSON data to /dev/null should be faster most of the time than handling an OCE anyway unless you're doing something unusual like `IAsyncEnumerable` serialization, where you would probably want to explicitly pass in `RequestAborted`.

Right now, the only way to get the fast pass is to pass in a token from a cancellable source and then not cancel it. Passing in `RequestAborted` explicitly mostly works for this, but that's annoying and could result in OCEs in your logs that you don't want.

@JamesNK @Tratcher do you have opinions on not passing in `RequestAborted` by default?